### PR TITLE
WIP: Enabling function-level XLA compilation

### PR DESCRIPTION
**Pull request by bgogul**
This is a WIP and needs some changes in TensorFlow as well as the CompilerRuntime.

Please take a look at xla.compile.swift to get a sense for the API to invoke xla compilation. The API is based on tracing. 

This PR also has a version of batch norm that uses the API above. It is not working yet:  I have got it compiling, but I am encountering some runtime failures. (Including it in this PR, just as FYI.)

### CODE REVIEW

File: XLABatchNorm/bnorm.swift [hunk 1]
```diff
@@ -0,0 +1,258 @@
+import TensorFlow
```
Comment (rxwei) [inline XLABatchNorm/bnorm.swift]: Use 4-space indentation.

**Comment by BradLarson** [comment]
Talking with Gogul, this is now obsolete, so I'll close it out.
